### PR TITLE
Resolve #708

### LINKED ISSUE #708: Rewrite discussions view methods to use Class based views
aristotle_mdr.views.discussions would be much cleaner using Class-based Views.

This should be a straight forward and small refactor, without any need to change templates or tests.

See issue #705 for more details on how to do this.

Read more about Class-based views here.
https://ccbv.co.uk/projects/Django/1.10/

### PR DESCRIPTION
**Pull request by itepifanio**
Passing View discussion.py from Function Based View to Class Based View.

### CODE REVIEW

File: aristotle_mdr/views/discussions.py [hunk 1]
```diff
@@ -87,19 +126,23 @@ def new(request):
...
-    post = get_object_or_404(MDR.DiscussionPost, pk=pid)
-    if not perms.user_in_workgroup(request.user, post.workgroup):
-        raise PermissionDenied
-    if post.closed:
-        messages.error(request, _('This post is closed. Your comment was not added.'))
-        return HttpResponseRedirect(reverse("aristotle:discussionsPost", args=[post.pk]))
-    if request.method == 'POST':
+
+
+class New_comment(FormView):
```
Comment (LegoStormtroopr) [inline aristotle_mdr/views/discussions.py]: Can you change these to CamelCase please?

File: venv/include/python3.5m [hunk 2]
```diff
@@ -0,0 +1 @@
+/usr/include/python3.5m
```
Comment (LegoStormtroopr) [inline venv/include/python3.5m]: Can you please remove this file, or explain what it is please?

**Comment by LegoStormtroopr** [comment]
Thanks for this! :)

It looks like this some PEP8 issues, but you can check this locally with tox using this command - `tox -e pep8`

I've also made a few comments. In #706 @DeKan has implemented the django-braces library for a LoginRequiredMixin. Would you mind doing the same on this branch as well, and removing the `login_required` decorateor from `aristotle_mdr.urls.aristotle`?

File: aristotle_mdr/views/discussions.py [hunk 3]
```diff
@@ -6,61 +6,97 @@
...
+    """
+    I put the decorator on urls.py, just in case to
+    sign a pattern, use this in the controller:
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        return super(All, self).dispatch(request, *args, **kwargs)
+    """    
+
+    def get(self, request, *args, **kwargs):
```
Comment (DeKaN) [inline aristotle_mdr/views/discussions.py]: I think, that you may override only get_context_data. Example: https://github.com/aristotle-mdr/aristotle-metadata-registry/pull/706/files#diff-7d9bac5f32f0b9848f898150653dda28R144

File: venv/pip-selfcheck.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"last_check":"2017-10-03T02:34:33Z","pypi_version":"9.0.1"}
```
Comment (LegoStormtroopr) [inline venv/pip-selfcheck.json]: Can you please remove this file, or explain what it is please?

**Comment by itepifanio** [comment]
- Fix override mentioned by @DeKaN. 
- PeP8 errors fixes (Is necessary fix the docs errors?).
- Removed extra files.
- Additioned Mixin to login required.

**Comment by LegoStormtroopr** [comment]
The docs tests are failing as the tox environment doesn't know it needs to install braces.

This can be fixed in two ways:
1. You can add `django-braces` into `aristotle_mr/setup.py` [like here](https://github.com/aristotle-mdr/aristotle-metadata-registry/pull/706/files#diff-2eeaed663bd0d25b7e608891384b7298)
2. Or you can pull from the develop branch, which includes the changes from #706 - but this might be more work for you.

I'm happy if you do option one, and get the tests working - I can address the merge issues.

Thanks again :)

**Comment by LegoStormtroopr** [comment]
To make sure this makes the next release I'll accept this pull, and have resolved the last few issues in #696.

Thanks very much!

File: aristotle_mdr/views/discussions.py [hunk 5]
```diff
@@ -110,66 +140,123 @@ def new_comment(request, pid):
             return HttpResponseRedirect(reverse("aristotle:discussionsPost", args=[new.post.pk]) + "#comment_%s" % new.id)
         else:
             return render(request, "aristotle_mdr/discussions/new.html", {"form": form})
-    else:
+
+    def get(self, request, *args, **kwargs):
         # It makes no sense to "GET" this comment, so push them back to the discussion
         return HttpResponseRedirect(reverse("aristotle:discussionsPost", args=[post.pk]))
```
Comment (DeKaN) [inline aristotle_mdr/views/discussions.py]: `post` is undefined here

Comment (LegoStormtroopr) [reply aristotle_mdr/views/discussions.py]: I've cleaned this up in a development branch I'm working on.
Should be merged in today.

**Pull request merged**